internal/pkg/agent/application/gateway/fleet/fleet_gateway.go

```diff
@@ -274,21 +274,34 @@ func (f *fleetGateway) gatherQueuedActions(ts time.Time) (queued, expired []flee
 	return queued, expired
 }
 
-func (f *fleetGateway) doExecute() (*fleetapi.CheckinResponse, error) {
+func (f *fleetGateway) executeCheckinWithRetries() (*fleetapi.CheckinResponse, error) {
 	f.backoff.Reset()
 
 	// Guard if the context is stopped by a out of bound call,
 	// this mean we are rebooting to change the log level or the system is shutting us down.
 	for f.bgContext.Err() == nil {
-		f.log.Debugf("Checkin started")
-		resp, err := f.execute(f.bgContext)
+		f.log.Debugf("Checking started")
+		resp, took, err := f.executeCheckin(f.bgContext)
 		if err != nil {
+			// Only update the local status on failure: https://github.com/elastic/elastic-agent/issues/1148
+			f.localReporter.Update(state.Degraded, fmt.Sprintf("checkin failed: %v", err), nil)
 			f.checkinFailCounter++
-			f.log.Errorf("Could not communicate with fleet-server checkin API will retry, error: %s", err)
+
+			// Report the first two failures at warn level as they may be recoverable with retries.
+			if f.checkinFailCounter <= 2 {
+				f.log.Warnw("Possible transient error during checkin with fleet-server, retrying",
+					"error.message", err, "request_duration", took, "failed_checkins", f.checkinFailCounter,
+					"retry_after", f.backoff.NextWait())
+			} else {
+				f.log.Errorw("Cannot checkin in with fleet-server, retrying",
+					"error.message", err, "request_duration", took, "failed_checkins", f.checkinFailCounter,
+					"retry_after", f.backoff.NextWait())
```

Comment:
Just asking, but did you check how the `retry_after` duration will be printed, if it'll be human readable or the milliseconds?

Reply:
It will be human readable according to https://pkg.go.dev/time#Duration.String since `NextWait()` returns a time.Duration.

Reply:
Actually I apparently need to manually call the `.String()` message if I want it to be human readable, otherwise the units are nanoseconds.

Reply:
I just added the `_ns` suffix to specify the units. The ns durations are easier to graph and work with in Kibana so I stuck with those.